File: tests/fixtures/edgecase_repo/javascript/src/module8.js
const { Service7 } = require("./module7");

class Base8 {
  constructor(name) {
    this.name = name;
  }
}

class Service8 extends Base8 {
  handle(payload) {
    this.validate(payload);
    return new Service7("x").handle(payload);
  }

  validate(payload) {
    return payload !== null;
  }
}

function helper8(value) {
  return value;
}

module.exports = { Service8 };

File: tests/fixtures/edgecase_repo/rust/src/module8.rs
use crate::module7::Service7;

pub struct Base8 {
    pub name: String,
}

pub struct Service8 {
    pub base: Base8,
}

pub trait Handler8 {
    fn handle(&self, payload: &str) -> bool;
}

impl Handler8 for Service8 {
    fn handle(&self, payload: &str) -> bool {
        self.validate(payload);
        true
    }
}

impl Service8 {
    fn validate(&self, payload: &str) {}
}

pub fn helper_8(value: &str) -> &str {
    value
}
